ARCv3: add extra perf cluster presence check by read of PCT_BUILD BCR

pavelvkozlov opened the pull request:
Read PCT_BUILD BCR and check perf cluster presence. Prevent read/write of AUX cluster resisters if they don't exist.

This commit in combination with QEMU commits:
https://github.com/foss-for-synopsys-dwc-arc-processors/qemu/commit/5e3a8972550b831db56803a7348dee8222a3a8c1
https://github.com/foss-for-synopsys-dwc-arc-processors/qemu/commit/a40d727a3fa54942fcc5af5a92b01e91cf8460d4
allows to run current ARCv3 Linux kernel with enabled CONFIG_PERF_EVENTS on QEMU.

pavelvkozlov committed:
ARCv3: add extra cluster presence check by read of CLUSTER_BUILD BCR

Read CLUSTER_BUILD BCR and check cluster presence before access to cluster
performance counter registers. Prevent read/write of AUX cluster CLNR_ADDR
and CLNR_DATA resisters if they don't exist.

Fixes: f269da7 ("ARCv3: Perf cluster driver fix")
Signed-off-by: Pavel Kozlov <pavel.kozlov@synopsys.com>

abrodkin reviewed (approved):
LGTM

abrodkin reviewed (approved):
LGTM

xxkent commented:
Please use CLUSTER_BUILD=0xCF register instead of ARC_REG_PCT_BUILD. 
Reading CLUSTER_BUILD we can decide whether cluster is presented in the system or not. If yes then we can get the access to CLNR_ADDR / CLNR_DATA registers.
ARC_REG_PCT_BUILD is a core PCT build configuration register, and it is not related to the cluster whose existence we are going to check.

pavelvkozlov commented:
I've added required changes, plz check.

pavelvkozlov committed:
ARCv3: perf_cluster.c replace spaces with tabs

Replase start line spaces with tabs for propper alignment (with 8 chars
tab setup). Squash this commit with f269da7 ("ARCv3: Perf cluster
driver fix") in the future.

Signed-off-by: Pavel Kozlov <pavel.kozlov@synopsys.com>

pavelvkozlov commented:
I've noticed that for proper alignment in 8 chars tab setup several lines are required to be tabbed (instead of spaces). I've added separate commit, to replace spaces. Let it be a reminder that we have enough files with this alignment issue that should be resolved in time of rebase.
